frontend/components/SendToGravePanel.tsx

```diff
@@ -0,0 +1,339 @@
+  const [showGhosts, setShowGhosts] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [canSubmit, setCanSubmit] = useState<boolean>(false);
+  const [inputTokenAddress, setInputTokenAddress] = useState<string>('');
+  const [tokenCheckMessage, setTokenCheckMessage] = useState<string>('');
+
+  const [tokenData, setTokenData] = useState<TokenData | undefined>(undefined);
+  const [debouncedTokenAddress, setDebouncedTokenAddress] =
+    useState(inputTokenAddress);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedTokenAddress(inputTokenAddress);
+    }, 500);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [inputTokenAddress]);
+
+  useEffect(() => {
+    setTokenCheckMessage('');
+    if (debouncedTokenAddress) {
+      processTokenData();
+    }
+  }, [debouncedTokenAddress]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputTokenAddress(event.target.value.toLowerCase());
+  };
+
+  const getTokenData = async () => {
+    let assetData;
+    try {
+      const wallet = await signer.getAddress();
+      assetData = await getLSPAssetBasicInfo(
+        provider,
+        inputTokenAddress,
+        wallet
+      );
+      setTokenData(assetData);
+      return assetData;
+    } catch (error: any) {
+      setTokenCheckMessage('Error fetching token data');
+      console.error(error);
+      toast({
+        title: `Error fetching token data. ${error.message}`,
+        status: 'error',
+        position: 'bottom-left',
+        duration: 9000,
+        isClosable: true,
+      });
+      setTokenCheckMessage('');
+      setCanSubmit(false);
+      return;
+    }
+  };
+
+  const processTokenData = async () => {
+    setCanSubmit(false);
+    setTokenCheckMessage('Checking...');
+    setTokenData(undefined);
+
+    const assetData = await getTokenData();
+    if (!assetData || !assetData.balance) {
+      setTokenCheckMessage('No balance for this Token');
+      setCanSubmit(false);
+      return;
+    }
+
+    if (Number(assetData.balance) === 0) {
+      setTokenCheckMessage(`Insufficient balance for ${assetData.symbol}`);
+      setCanSubmit(false);
+      return;
+    }
+
+    const tokenType = assetData?.tokenType;
+    const readableTokenAmount = parseFloat(
+      ethers.utils.formatUnits(assetData?.balance, assetData?.decimals)
+    ).toFixed(
+      tokenType === LSP4_TOKEN_TYPES.TOKEN ? Number(assetData?.decimals) : 0
+    );
+
+    const roundedTokenAmount = parseFloat(readableTokenAmount).toFixed(
+      getEnoughDecimals(Number(readableTokenAmount))
+    );
+
+    setTokenCheckMessage(
+      `Token balance: ${assetData?.symbol} ${roundedTokenAmount}`
+    );
+    setCanSubmit(true);
+  };
+
+  const removeTokenFromAllowList = async () => {
+    const LSP1GraveForwarderContract = new ethers.Contract(
+      networkConfig.universalGraveForwarder,
+      LSP1GraveForwarderAbi.abi,
+      signer
+    );
```

Comment:
minor but would be good to use the typechain version of this to get type saftey / checks

```
      const LSP1GraveForwarderContract = LSP1GraveForwarder__factory.connect(
        networkConfig.universalGraveForwarder,
        signer
      );
```

Reply:
done